test/nbrowser/ActiveUserList.ts

```diff
@@ -0,0 +1,120 @@
+import {UserAPI} from 'app/common/UserAPI';
+import * as gu from 'test/nbrowser/gristUtils';
+import {server, setupTestSuite} from 'test/nbrowser/testUtils';
+
+import {assert} from 'chai';
+import {driver} from 'mocha-webdriver';
+
+interface Window {
+  handle: string;
+}
+
+const UserOwner: gu.TestUser = 'user1';
+const User2: gu.TestUser = 'user2';
+const User3: gu.TestUser = 'user3';
+const USER_PRESENCE_MAX_USERS = 6;
+
+describe('ActiveUserList', async function() {
+  this.timeout('120s');
+  const extraWindowHandles: string[] = [];
+
+  // Needs to be registered before 'setupTestSuite' to ensure windows are closed before its afterAll hook
+  after(async () => {
+    for (const handle of extraWindowHandles) {
+      await driver.switchTo().window(handle);
+      await driver.close();
+    }
+    // Makes sure the correct window is selected for the cleanup scripts in setupTestSuite
+    await switchToWindow(mainWindow);
+  });
+
+  const cleanup = setupTestSuite();
+  let mainWindow: Window;
+  let docId: string;
+  let ownerSession: gu.Session;
+  let ownerApi: UserAPI;
+  const windowsByUser: Map<gu.TestUser, Window[]> = new Map();
+
+  before(async () => {
+    process.env.GRIST_ENABLE_USER_PRESENCE = 'true';
+    process.env.GRIST_USER_PRESENCE_MAX_USERS = USER_PRESENCE_MAX_USERS.toFixed(0);
+    await server.restart();
```

Comment:
This doesn't get cleaned up after the suite finishes, right?

`withEnvironmentSnapshot` from `test/nbrowser/gristUtils.ts` can help with mutations of the environment.